vm/src/stdlib/syslog.rs

```diff
@@ -0,0 +1,102 @@
+pub(crate) use syslog::make_module;
+
+#[pymodule(name = "syslog")]
+mod syslog {
+    use crate::builtins::pystr::{PyStr, PyStrRef};
+    use crate::function::{OptionalArg, OptionalOption};
+    use crate::pyobject::PyValue;
+    use crate::utils::ToCString;
+    use crate::vm::VirtualMachine;
+    use crate::{PyObjectRef, PyResult, TryFromObject};
+
+    #[pyattr]
+    use libc::{
+        LOG_ALERT, LOG_AUTH, LOG_CONS, LOG_CRIT, LOG_DAEMON, LOG_DEBUG, LOG_EMERG, LOG_ERR,
+        LOG_INFO, LOG_KERN, LOG_LOCAL0, LOG_LOCAL1, LOG_LOCAL2, LOG_LOCAL3, LOG_LOCAL4, LOG_LOCAL5,
+        LOG_LOCAL6, LOG_LOCAL7, LOG_LPR, LOG_MAIL, LOG_NDELAY, LOG_NEWS, LOG_NOTICE, LOG_NOWAIT,
+        LOG_ODELAY, LOG_PID, LOG_SYSLOG, LOG_USER, LOG_UUCP, LOG_WARNING,
+    };
+
+    #[cfg(not(target_os = "redox"))]
+    #[pyattr]
+    use libc::{LOG_AUTHPRIV, LOG_CRON, LOG_PERROR};
+
+    fn get_argv(vm: &VirtualMachine) -> PyResult<PyStrRef> {
+        if let Some(argv) = vm.state.settings.argv.first() {
+            if !argv.is_empty() {
+                return Ok(PyStr::from(match argv.find('\\') {
+                    Some(value) => argv[value..].to_owned(),
+                    None => argv.to_owned(),
+                })
+                .into_ref(vm));
+            }
+        }
+        Err(vm.new_value_error("empty sys.argv was given".to_owned()))
+    }
+
+    #[derive(FromArgs)]
+    struct OpenLogArgs {
+        #[pyarg(any, optional)]
+        ident: OptionalOption<PyStrRef>,
+        #[pyarg(any, optional)]
+        logoption: OptionalArg<i32>,
+        #[pyarg(any, optional)]
+        facility: OptionalArg<i32>,
+    }
+
+    #[pyfunction]
+    fn openlog(args: OpenLogArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let logoption = args.logoption.unwrap_or(0);
+        let facility = args.facility.unwrap_or(LOG_USER);
+        let ident = args
+            .ident
+            .flatten()
+            .unwrap_or(get_argv(vm)?)
+            .to_cstring(vm)?;
+
+        unsafe { libc::openlog(ident.as_ptr(), logoption, facility) };
+        Ok(())
+    }
+
+    #[derive(FromArgs)]
+    struct SysLogArgs {
+        #[pyarg(positional)]
+        priority: PyObjectRef,
+        #[pyarg(positional, optional)]
+        message_object: OptionalOption<PyStrRef>,
+    }
+
+    #[pyfunction]
+    fn syslog(args: SysLogArgs, vm: &VirtualMachine) -> PyResult<()> {
+        let (priority, msg) = match args.message_object.flatten() {
+            Some(s) => (i32::try_from_object(vm, args.priority)?, s),
+            None => (LOG_INFO, PyStrRef::try_from_object(vm, args.priority)?),
+        };
+
+        let (cformat, cmsg) = ("%s".to_cstring(vm)?, msg.to_cstring(vm)?);
+        unsafe { libc::syslog(priority, cformat.as_ptr(), cmsg.as_ptr()) };
```

Comment:
I haven't been able to find a clear reason why the problem you have exists but it may be because you just call `syslog` here. In `CPython`, `syslog` calls `openlog` if it hasn't already done so (see `S_log_open`) and that might have something to do with it (I *think* `syslog` will try and call `openlog` again in some cases but I'm not feeling like looking into the implementation of `syslog` in `libc`).

Try following along with what they do and see if this solves it. 

Note: this could be done in a following PR so as to not stall the whole module being included. The other comment I left should be addressed in this PR, though (and it should be easier to address).

Reply:
There also exists problem when I called `syslog` after make connection to syslogger by calling `openlog`. As global variable `S_log_open` in cpython is only for make guarantee there exists connection to syslogger, I think it will not resolve this issue 😢
By the way, a mechanism something like global variable in cpython should be implemented.

Reply:
one of OnceCell, lazy_static or LocalKey will be helpful for static variable

Reply:
@Snowapril  really not sure. Maybe it has something to do with how `syslog` is implemented, a double call to `openlog` might mess things up. Either way, I think this is a relatively minor bug for now so it can be addressed at some later point (I would like a `TODO:` comment though so we don't forget about it).